core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -865,9 +865,13 @@ class QualificationAppInfo(
         logWarning(s"Application $appId: Cluster with variable executor cores detected. " +
           s"Using maximum value.")
       }
+      // Group by host name, find max executors per host, and number of unique hosts
+      val groupedHosts = activeHosts.groupBy(identity)
```

Comment:
so we don't want to just use activeHosts, we want to use all hosts to determine execs per node.  That will be more likely right in case as an application finishes, if it has dynamic allocation on, it could release executors and thus the final active count could be inaccurate.

Put the number of executor nodes back to the active hosts for now but really we should try to do some timelining of this to see what was the max number in use at any time, but filed https://github.com/NVIDIA/spark-rapids-tools/issues/1121 to followup with this.  Put it back to what it was for now.

Reply:
also it would be good to get a event log where dynamic allocation is disabled and the number of execs/hosts change over time. This is relatively easy in interactive, where you just run something then wait for the execs to idle timeout. then maybe run something again, maybe smaller.

Reply:
> we don't want to just use activeHosts, we want to use all hosts to determine execs per node.

Updated the code to use all hosts.

> Put the number of executor nodes back to the active hosts for now 

Reverted to the original logic.

Reply:
Created a scenario for this (I think this meant dynamic allocation to be enabled):
> get a event log where dynamic allocation is disabled and the number of execs/hosts change over time
<img width="1239" alt="image" src="https://github.com/NVIDIA/spark-rapids-tools/assets/13639815/efcb5099-ca45-468c-a97e-963b172db74c">


**Steps:**
1. Created a dataproc cluster with 4 `n1-standard-16` worker nodes.
1. Start a spark shell with ` --conf spark.dynamicAllocation.enabled=true  --conf spark.executor.instances=8 --conf spark.executor.cores=8`
2. Run a large spark application. UI shows  all 8 executors are active (2 per worker node).
3. Wait for 30min. 7 executors are dead.
4. Run a small spark application. UI shows it 1 executor is active.

Now since we are calculating execs from all hosts, we will get correct num of exec/hosts as 2 instead of 1.